### PR TITLE
Upgrade OpenTelemetry

### PR DESCRIPTION
**Pull request by atoulme**
Signed-off-by: Antoine Toulme <antoine@lunar-ocean.com>

## PR description
Upgrades OpenTelemetry to the latest version, and make sure we configure the SDK with environment variables.

## Documentation

- [X] I thought about documentation and added the `doc-change-required` label to this PR if
    [updates are required](https://wiki.hyperledger.org/display/BESU/Documentation).

## Changelog

- [x] I thought about the changelog and included a [changelog update if required](https://wiki.hyperledger.org/display/BESU/Changelog).

### CODE REVIEW

**Comment by macfarla** [comment]
@atoulme - missing SPDX header 
> Files without headers: /home/circleci/project/metrics/core/src/main/java/org/hyperledger/besu/metrics/opentelemetry/DebugMetricReader.java

File: metrics/core/src/test/java/org/hyperledger/besu/metrics/opentelemetry/OpenTelemetryMetricsSystemTest.java [hunk 1]
```diff
@@ -248,17 +262,4 @@ public void returnsPrometheusMetricsWhenEnabled() {
...
-    final MetricsConfiguration metricsConfiguration =
-        MetricsConfiguration.builder()
-            .enabled(false)
-            .pushEnabled(true)
-            .protocol(OPENTELEMETRY)
-            .build();
-    final MetricsSystem localMetricSystem = MetricsSystemFactory.create(metricsConfiguration);
-
-    assertThat(localMetricSystem).isInstanceOf(OpenTelemetrySystem.class);
-  }
```
Comment (macfarla) [inline metrics/core/src/test/java/org/hyperledger/besu/metrics/opentelemetry/OpenTelemetryMetricsSystemTest.java:263]: do we not need this test?

Comment (atoulme) [reply metrics/core/src/test/java/org/hyperledger/besu/metrics/opentelemetry/OpenTelemetryMetricsSystemTest.java:263]: Doesn't feel like it tests much.

**Review by jflo — APPROVED** [review]

**Comment by macfarla** [comment]
@atoulme there's still some ATs failing on this one

**Comment by atoulme** [comment]
Sorry, long time. I got back and fixed the AT locally, let's see it.

File: metrics/core/src/test/java/org/hyperledger/besu/metrics/opentelemetry/OpenTelemetryMetricsSystemTest.java [hunk 2]
```diff
@@ -50,23 +50,24 @@ public class OpenTelemetryMetricsSystemTest {
...
-  public void shouldCreateObservationFromCounter() {
+  public void shouldCreateObservationFromCounter() throws InterruptedException {
     final Counter counter = metricsSystem.createCounter(PEERS, "connected", "Some help string");
 
     counter.inc();
     assertThat(metricsSystem.streamObservations())
         .containsExactly(new Observation(PEERS, "connected", 1L, emptyList()));
 
     counter.inc();
+    Thread.sleep(1000);
```
[the remark this answers is not in the document]
Comment (atoulme) [reply metrics/core/src/test/java/org/hyperledger/besu/metrics/opentelemetry/OpenTelemetryMetricsSystemTest.java]: Asynchronous collector managed by otel. I removed the absolute sleeps and use polling with smaller sleep intervals as a workaround.

File: metrics/core/src/main/java/org/hyperledger/besu/metrics/opentelemetry/MetricsOtelPushService.java [hunk 3]
```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright ConsenSys AG.
```
[the remark this answers is not in the document]
Comment (atoulme) [reply metrics/core/src/main/java/org/hyperledger/besu/metrics/opentelemetry/MetricsOtelPushService.java]: fixed thanks!

**Pull request merged**